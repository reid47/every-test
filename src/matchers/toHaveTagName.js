import { formatValue } from './common';

export default function toHaveTagName(received, expectedTagName) {
  const actualTagName = received.tagName.toLowerCase();
  const pass = actualTagName === expectedTagName.toLowerCase();

  const message = pass
    ? `Expected element not to have tag name ${formatValue(expectedTagName)}.`
    : `Expected element to have tag name ${formatValue(
        expectedTagName
      )}, but actual tag name was ${formatValue(actualTagName)}.`;

  return { pass, message };
}
