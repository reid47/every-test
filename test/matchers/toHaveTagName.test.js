import toHaveTagName from '../../src/matchers/toHaveTagName';

describe('toHaveTagName', () => {
  test('passing', () => {
    const el = document.createElement('div');

    const result = toHaveTagName(el, 'div');
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('div');

    const result = toHaveTagName(el, 'button');
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
