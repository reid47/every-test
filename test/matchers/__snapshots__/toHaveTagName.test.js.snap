// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHaveTagName failing 1`] = `"Expected element to have tag name \\"button\\", but actual tag name was \\"div\\"."`;

exports[`toHaveTagName passing 1`] = `"Expected element not to have tag name \\"div\\"."`;
